opent2t.p.outlet/com.wink.binaryswitch/js/thingTranslator.js

```diff
@@ -0,0 +1,158 @@
+        }
+        else {
+            return this.last_reading[state];
+        }
+    }
+}
+
+// Helper method to convert the provider schema to the platform schema.
+function providerSchemaToPlatformSchema(providerSchema, expand) {
+    var stateReader = new StateReader(providerSchema.desired_state, providerSchema.last_reading);
+
+    var powered = stateReader.get('powered');
+
+    var power = {
+        href: '/power',
+        rt: ['oic.r.switch.binary'],
+        if: ['oic.if.a', 'oic.if.baseline']
+    };
+    
+    if (expand) {
+        power.id = 'power';
+        power.value = powered;
+    }
+
+    return {
+        opent2t: {
+            schema: 'opent2t.p.outlet',
+            translator: 'opent2t-translator-com-wink-binaryswitch',
+            controlId: deviceId
+        },
+        pi: providerSchema['uuid'],
+        mnmn: providerSchema['device_manufacturer'],
+        mnmo: providerSchema['manufacturer_device_model'],
+        n: providerSchema['name'],
+        rt: ['opent2t.p.outlet'],
+        entities: [
+            {
+                rt: ['oic.d.smartplug'],
+                di: lightDeviceDi,
+                resources: [
+                    power
+                ]
+            }
+        ]
+    };
+}
+
+// Helper method to convert the translator schema to the device schema.
+function resourceSchemaToProviderSchema(resourceId, resourceSchema) {
+
+    // build the object with desired state
+    var result = { 'desired_state': {} };
+    var desired_state = result.desired_state;
+
+    if ('power' === resourceId) {
+        desired_state['powered'] = resourceSchema.value;
+    }
+
+    return result;
+}
+
+function findResource(schema, di, resourceId) {
+    var entity = schema.entities.find((d) => {
+        return d.di === di;
+    });
+
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    return resource;
+}
+
+var deviceId;
+var deviceType = 'binary_switches';
+var winkHub;
+
+// Each device in the platform has its own static identifier
+const lightDeviceDi = 'F85B0738-6EC0-4A8B-A95A-503B6F2CA0D8';
+
+// This translator class implements the 'opent2t.p.outlet' interface.
+class Translator {
+
+    constructor(deviceInfo) {
+        console.log('Wink Binary Switch initializing...');
+
+        validateArgumentType(deviceInfo, "deviceInfo", "object");
+        
+        deviceId = deviceInfo.deviceInfo.id;
+        winkHub = deviceInfo.hub;
+
+        console.log('Wink Binary Switch initializing...Done');
+    }
+
+    // exports for the entire schema object
+
+    // Queries the entire state of the binary switch
+    // and returns an object that maps to the json schema opent2t.p.outlet
+    get(expand) {
+        return winkHub.getDeviceDetailsAsync(deviceType, deviceId)
+            .then((response) => {
+                return providerSchemaToPlatformSchema(response.data, expand);
+            });
+    }
+
+    getDeviceResource(di, resourceId) {
```

Comment:
I implemented it this way because I didn't see much reason to end up with a bunch of getDeviceResource{X}(di) functions defined that all did essentially the same thing. At the end of the day, we need to define the contract that we expect between callers of OpenT2T and the translators.

Perhaps @arjun-msft  or @avneet723 could provide their thoughts.